iothub/service/src/DigitalTwin/Serialization/ComponentMetadata.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Devices.Serialization
+{
+    /// <summary>
+    /// An optional, helper class for deserializing a digital twin.
+    /// The $metadata class on a component of a <see cref="BasicDigitalTwin"/>.
+    /// </summary>
+    public class ComponentMetadata
+    {
+        /// <summary>
+        /// Model-defined writable properties' request state.
+        /// </summary>
+        /// <remarks>For convenience, the value of each dictionary object can be turned into an instance of <see cref="WritableProperty"/>.</remarks>
+        [JsonExtensionData]
+        public IDictionary<string, object> WriteableProperties { get; } = new Dictionary<string, object>();
```

Comment:
I posted a question in teams about choosing Writeable or Writable, both valid spellings. I realized we have it different in a few places.